### PR TITLE
added some configuration instructions to readme

### PR DESCRIPTION
**Pull request by Thomas-Lemoine**
some things left:
- I don't exactly understand how airtable stuff works, and there's no instructions for making that work in the readme
- `--rebuild` was removed, not sure if we ought to remove it from the readme or make a similar thing available in some way. like either dropping the articles from a source before going through them or upserting for each or wtv.

I didn't test these: agisf arxiv indices special_docs youtube, since they were either very slow or didn't work at the time

Some things I may have changed that were best left unchanged, if so lmk

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,146 +64,206 @@ Additional keys may be available depending on the source document.
...
+### AlignmentDataset class
 
-    python main.py update_metadata <input spreadsheet id> <input sheet name> <output spreadsheet id>
+This is the main workhorse for processing datasets. The basic idea is that it provided a list of items to be processed, and after processing a given item, appends it to the appropriate jsonl file, where each line of the file is a JSON object with all the data. The `AlignmentDataset` class has various methods that can be implemented to handle various cases. A few assumptions are made as to the data it will use, i.e.:
```
Comment (mruwnik) [inline README.md:240]: this should be overhauled... :D  Though mainly by removing the jsonl info and adding that it saves it to MySQL

File: README.md [hunk 2]
```diff
@@ -64,146 +64,206 @@ Additional keys may be available depending on the source document.
...
 
-which for the current documents would be:
+* `self.data_path` is where data will be written to and read from - by default it's the `data/` directory
+* `self.raw_data_path` is where downloaded files etc. should go - by default it's the `data/raw` directory
+* `self.files_path` is where data to be processed is expected to be. This is used e.g. when a collection of html files are to be processed
```
Comment (mruwnik) [inline README.md:244]: these also aren't really used now

File: README.md [hunk 3]
```diff
@@ -64,146 +64,206 @@ Additional keys may be available depending on the source document.
...
 
-    python main.py update_metadata 1pgG3HzercOhf4gniaqp3tBc3uvZnHpPhXErwHcthmbI special_docs.csv 1l3azVJVukGAvZPgg0GyeqiaQe8bEMZvycBJaA8cRXf4
+The `AlignmentDataset` is a dataclass, so it has a couple of settings that control it:
 
-#### Google API
+* `name` - this is a string that identifies the dataset, i.e. 'lesswrong'
+* `done_key` - used to check if a given item has already been processed. This is a key in the JSON object that gets written to the output file - any subsequent entries with the same value for that key will be skipped
```
Comment (mruwnik) [inline README.md:249]: this should be removed

File: README.md [hunk 4]
```diff
@@ -64,146 +64,206 @@ Additional keys may be available depending on the source document.
...
- * Enable the Google Sheets API for your project at https://console.cloud.google.com/apis/api/sheets.googleapis.com/metrics?project=<your project id>
- * Enable the Google Drive API for your project at https://console.cloud.google.com/apis/api/drive.googleapis.com/metrics?project=<your project id>
- An alternative to this step is that when running the program without these enabled, an exception will be raised telling you how to enable it - you can then just open the link in the exception message
-4. Navigate to the "Credentials" section
-5. Click on "Create Credentials" and select "Service Account"
-6. Fill in the required information for the service account
-7. On the "Create key" page, select the JSON key type and click "Create"
-8. The JSON file containing your credentials will be downloaded -> save as credentials.json in the folder from which you're running the code
+1. `self.setup()` - any instance level initialization stuff should go here, e.g. fetching zip files with data
+2. `self._load_outputted_items()` - go through `self.jsonl_path` and construct a set of the `self.done_key` values of each item - this is used to skip items that have already been processed
```
Comment (mruwnik) [inline README.md:255]: this should be changed to say that it gets the urls of previously processed items from the database

File: README.md (hunk 1 above)
Comment (Thomas-Lemoine) [reply README.md:240]: ah, good catch

File: README.md (hunk 2 above)
Comment (Thomas-Lemoine) [reply README.md:244]: they seem only to be used by datasets.py and by agentmodels.py

File: README.md (hunk 3 above)
Comment (Thomas-Lemoine) [reply README.md:249]: Do I just keep "* `done_key` - used to check if a given item has already been processed."?

File: README.md (hunk 4 above)
Comment (Thomas-Lemoine) [reply README.md:255]: Is my understanding wrong that this still uses done_key and not strictly urls? If not, how about this:
```
2. `self._load_outputted_items()` - go through articles in the database, loads the value of their `self.done_key`, and skips items that have a `self.done_key` already present in the db
```

Comment (Thomas-Lemoine) [reply README.md:255]: """
2. `self._load_outputted_items()` - go through articles in the database, loads the value of their `self.done_key`, outputs a simplefied version of these strings. The purpose for this is to then compare new items against these _outputted_items, and skip items that are already in the database
"""
is a bit more explicit

File: README.md (hunk 3 above)
Comment (mruwnik) [reply README.md]: yup. Maybe with a note that the default is to just check the `url` field

**Review by mruwnik — APPROVED** [review]

**Pull request merged**